### PR TITLE
Improved migration utils; recommendation data issues addressed

### PR DESCRIPTION
**Pull request by hero101**
### Describe the background of your pull request

What does your pull request do? Does it solve a bug (which one?), add a feature?

### Additional context

Add any other context

### Governance 

- [ ] Documentation is added
- [ ] Test cases are added or updated

By submitting this pull request I confirm that:
- I've read the contributing document https://github.com/alkem-io/.github/blob/master/CONTRIBUTING.md.
- I've read and understand the Code of Conduct https://github.com/alkem-io/.github/blob/master/CODE_OF_CONDUCT.md.
- I understand that any contributions or suggestions I made may make it into the actual code. I've read the License 
     https://github.com/alkem-io/Coordination/blob/master/LICENSE.
- I understand that submitting the pull request requires agreeing to and signing the contributor license agreement.

### CODE REVIEW

**Comment by hero101** [comment]
To be merged with  https://github.com/alkem-io/client-web/pull/3922

File: src/migrations/types/callout-state.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export enum CalloutState {
```
Comment (valentinyanakiev) [inline src/migrations/types/callout-state.ts]: Please move this state as part of the migration using it. The state is a snapshot in time and if the states change, you change this enum, and then re-running the migrations will change its original behavior, which is not intended.

File: src/migrations/types/callout-visiblity.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export enum CalloutVisibility {
```
Comment (valentinyanakiev) [inline src/migrations/types/callout-visiblity.ts]: Same as previous one.

**Review by valentinyanakiev — CHANGES_REQUESTED** [review]
Two minor comments to be fixed, other than that migration worked on production data.

**Review by valentinyanakiev — APPROVED** [review]

**Pull request merged**